docs/cookbook.rst

```diff
@@ -0,0 +1,138 @@
+The Nox Cookbook
+================
+
+The What?
+---------
+
+A lot of people and a lot of projects use ``nox`` for their python automation powers.
+
+Some of these sessions are the classic "run pytest and linting", some are more unique and more interesting!
+
+The nox cookbook is a collection of these such sessions.
+
+``nox`` is super easy to get started with, and super powerful right out of the box. But when things get complex or you want to chain together some more powerful tasks, often the only examples can be found hunting around GitHub for novel ``nox`` sessions.
+
+The kind of sessions that make you think "I didn't know you could do that!"
+
+This cookbook is intended to be a centralized, community-driven repository of awesome ``nox`` sessions to act as a source of inspiration and a reference guide for nox's users. If you're doing something cool with ``nox``, why not add your session here?
+
+
+Contributing a Session
+----------------------
+
+Anyone can contribute sessions to the cookbook. However, there are a few guiding principles you should keep in mind:
+
+* Your session should be interesting or unique, it should do something out of the ordinary or otherwise interesting.
+* You should explain briefly what it does and why it's interesting.
+* You should include the full session as a code block, if it's too large, try to include a minimal example. Bonus points if a user can copy and paste it into their ``noxfile.py`` and run it straight away!
+
+
+And now on with the recipes...
+
+Recipes
+-------
+
+Instant Dev Environment
+^^^^^^^^^^^^^^^^^^^^^^^
+
+A common sticking point in contributing to python projects (especially for beginners) is the problem of wrangling virtual environments and installing dependencies.
+
+Enter the ``dev`` nox session:
+
+.. code-block:: python
+
+    import os
+    
+    import nox
+
+    # It's a good idea to keep your dev session out of the default list
+    # so it's not run twice accidentally
+    nox.options.sessions = [<sessions other than dev>]
```

Comment:
You can use an ellipsis here and it will remain valid python code.

Reply:
Always wondered what ellipsis was used for! thanks!